pennylane/_device.py

```diff
@@ -222,14 +226,22 @@ def execute(self, queue, observables, parameters={}):
 
             for obs in observables:
                 if obs.return_type == "expectation":
-                    results.append(self.expval(obs.name, obs.wires, obs.parameters))
+                    if obs.name == "Tensor":
+                        # Pass the full tensor object rather than just names
+                        results.append(self.expval(obs, obs.wires, obs.parameters))
+                    else:
+                        results.append(self.expval(obs.name, obs.wires, obs.parameters))
```

Comment:
@quantshah, I have a suggestion here that will avoid the `if` statement, and treat a `Tensor` object like any other observable.

For example, consider the following tensor:
```python
obs = Tensor()
obs += PauliX(wires=0)
obs += PauliZ(wires=1)
obs += Hermitian(A, wires=[2, 3])
```
With each observable appended, we could just have the class do
```python
    def append(self, op):
        """Add operations to a tensor list"""
        self.ops.append(op)
        self.name.append(op.name)
        self.wires.append(op.wires)
        self.params.append(op.params)

    __iadd__ = append
```

then we would have
```python
>>> obs.name
['PauliX', 'PauliZ', 'Hermitian']
>>> obs.wires
[[0], [1], [2, 3]]
>>> obs.parameters
[[], [], [A]]
```

It would then be up to the plugin to recognize that if `isinstance(obs.name, list)`, then the operation is a tensor (this could be part of the developer API).